src/jit/lower.cpp

```diff
@@ -2742,6 +2742,7 @@ void Lowering::InsertPInvokeCallProlog(GenTreeCall* call)
 
         comp->fgMorphTree(helperCall);
         BlockRange().InsertBefore(insertBefore, LIR::SeqTree(comp, helperCall));
+        LowerNode(helperCall); // helper call is inserted before current node and should be lowered here.
```

Comment:
I think we also need to do this here: https://github.com/dotnet/coreclr/pull/7605/files#diff-f36b3b4ad6feb9469b14f5ba2047cf74R2888